### ISSUE TITLE
Predefined positions set by the user

### ISSUE THREAD

**Issue by dazito** [skjolber/3d-bin-container-packing#97]
A nice feature is to allow some boxes to be positioned beforehand in some specific places (pre defined by the user) and then pack the remaining boxes in the remaining free spaces.

Example, 
- We have a container 50x50x50
- We specify that in position (x,y,z) 10, 0, 10 we already have a box with dimension 5, 5, 5
- We have 2 boxes to be packed with dimensions 10, 10, 10 and 15, 15, 15
- Now we ask the library to pack the two boxes above but to take in consideration that position 10, 0, 10 is already occupied with a box with dimension 5, 5, 5 and therefore to not place these two boxes there.

Did I make myself clear?

**Comment by skjolber** [comment]
Yes. This might be incorporated into the 'free space' calculations somehow.

**Comment by skjolber** [comment]
Perhaps here: https://github.com/skjolber/3d-bin-container-packing/blob/master/src/main/java/com/github/skjolberg/packing/LargestAreaFitFirstPackager.java#L296

**Comment by dazito** [comment]
I was debugging this part of the code to fully understand the logic and I have a few questions regarding `getFreespaces` method.

1. The variable `used`, of type `Box`, is the box we want to place inside the container, right?
2. The output `Space[] freeSpaces` is a pair of coordinates representing two free areas where one pair member is the "starting" coordinate and the other the "end" coordinate, correct?
3. Would it make sense if we pass in to `getFreespaces` another parameter (a list of `Box`es) representing the user defined occupied spaces and then let the `getFreespaces` calculate the free spaces left? But this time there may be more than 2 free spaces as seen below, right?

```
+-------------+----------------X----------------------+
|             |                X                      |
|             |                X                      |
|      A      |                X         3            |
|             |                X                      |
|             |                X                      |
|             |                X                      |
X-------------X                X-----------XXXXXXXXXXXX
X             X                |           |          |
X             X                |           |          |
X             X        2       |     B     |          |
X             X                |           |          |
X      1      X                |           |          |
X             X                |           |          |
X             X                X-----------XXXXXXXXXXXX
X             X                X                      |
X             X                X                      |
X             X                X           4          |
X             X                X                      |
X             X                X                      |
XXXXXXXXXXXXXXX----------------X----------------------+

```
A => Box to be placed
B => Locked space - the user set this area as not free

1, 2, 3 and 4 are a possible set of free spaces

**Comment by skjolber** [comment]
I think this is a good idea.

**Comment by skjolber** [comment]
@dazito this can work in the new version 2
